src/input/plugins/CdioParanoiaMusicBrainzTags.hxx

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include "lib/curl/Init.hxx"
+#include "lib/curl/Headers.hxx"
+#include "lib/curl/Request.hxx"
+#include "lib/curl/StringHandler.hxx"
+#include "input/RemoteTagScanner.hxx"
+#include "tag/Builder.hxx"
+#include "tag/Tag.hxx"
+
+#include <map>
+#include <set>
+
+class MusicBrainzCDTagCache
+{
+public:
+	MusicBrainzCDTagCache (EventLoop &event_loop)
+	: curl(event_loop)
+	{
+	}
+
+	virtual ~MusicBrainzCDTagCache ()
+	{
+		clearTracks();
+	}
+
+public:
+	struct TrackInfo
+	{
+		int trackNum = -99;
+		std::string title;
+		std::string artist;
+		std::string firstReleaseDate;
+		std::string albumTitle;
+		std::string albumDate;
+		std::string albumArtist;
+		std::string albumGenre;
+		int duration = 0;
+	};
+
+public:
+	class Listener
+	{
+	public: // MusicBrainzCDTagCache::Listener
+		virtual void setTags (MusicBrainzCDTagCache::TrackInfo& trackInfo) = 0;
+	};
+
+public:
+	void requestTags (std::string_view& uri, std::string device_, Listener *listener);
+	void cancelRequest (Listener* listener);
+	bool insertedCdChanged ();
+	void clearTracks ();
+	void requestMusicBrainzTags();
+	bool makeTrackInfoFromXml (std::string& body);
+	void callListeners ();
+
+public:
+	static void deleteInstance ()
+	{
+		if (instance != nullptr)
+			delete instance;
+		instance = nullptr;
+	}
+
+	static void createInstance (EventLoop &event_loop)
+	{
+		instance = new MusicBrainzCDTagCache(event_loop);
+	}
+
+	static MusicBrainzCDTagCache* getInstance ()
+	{
+		return instance;
+	}
+
+private:
+	std::string lastCdId;
+	std::map<int, TrackInfo> tracks;
+
+	static MusicBrainzCDTagCache *instance;
+
+	std::map<int, std::set<Listener*> > listeners;
```

Comment:
What's this, why is it a std::map, what's the index and why is there a space between the two `>`?